tensorflow/python/compiler/tensorrt/test/trtengineop_input_output_shape_test.py

```diff
@@ -0,0 +1,108 @@
+from tensorflow.python.framework import constant_op
+from tensorflow.python.framework import dtypes
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import gen_math_ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.platform import test
+
+import numpy as np
+import tensorflow as tf
+from tensorflow.python.compiler.tensorrt import trt_convert as trt
+from tensorflow.python.saved_model import signature_constants
+from tensorflow.python.saved_model import tag_constants
+
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import nn
+
+
+class TRTEngineOpInputOutputShapeTest(trt_test.TfTrtIntegrationTestBase):
+  """Testing conversion of Conv2D (data_format=NCHW) in TF-TRT conversion."""
+
+  def GraphFn(self, inp):
+    np.random.seed(1234)
+    b = array_ops.squeeze(inp, axis=[2])
+    c = nn.relu(b)
+    d1 = c + c
+    d2 = math_ops.reduce_sum(d1)
+    return d1, d2
+    # return d1
+
+  def GetParams(self):
+    # TODO(aaroey): test graph with different dtypes.
+    return self.BuildParams(self.GraphFn, dtypes.float32, [[1, 2, 1, 4]],
+                            [[1, 2, 4], []])
+
+  def _GetInferGraph(self, *args, **kwargs):
+      trt_saved_model_dir = super(
+          TRTEngineOpInputOutputShapeTest, self
+      )._GetInferGraph(*args, **kwargs)
+
+      def get_func_from_saved_model(saved_model_dir):
+          saved_model_loaded = tf.saved_model.load(
+              saved_model_dir, tags=[tag_constants.SERVING])
+          graph_func = saved_model_loaded.signatures[
+              signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY]
+          return graph_func, saved_model_loaded
+
+      func, loaded_model = get_func_from_saved_model(trt_saved_model_dir)
+
+      print("Input tensor", func.inputs)
+      if isinstance(func.inputs[0].shape, tf.TensorShape):
+          input_shape = func.inputs[0].shape
+      else:
+          input_shape = tf.TensorShape(func.inputs[0].shape)
+      assert (input_shape == tf.TensorShape((None, 2, 1, 4)))
+
+      print("Output tensor", func.outputs)
+
+      if isinstance(func.outputs[0].shape, tf.TensorShape):
+          output_shapes = [output.shape for output in func.outputs]
+      else:
+          output_shapes = [
+              tf.TensorShape(output.shape) for output in func.outputs]
+
+      assert (output_shapes[0] == tf.TensorShape(dims=(None, 2, 4)))
+      assert (output_shapes[1] == tf.TensorShape(dims=()))
+
+      return trt_saved_model_dir
+
+  def ExpectedEnginesToBuild(self, run_params):
+    """Return the expected engines to build."""
+    # return ["YOLO"]
+    return ["TRTEngineOp_0"]
+
+    # return {
+    #   'TRTEngineOp_0': [
+    #       'combined_nms/CombinedNonMaxSuppression',
+    #       'max_output_size_per_class', 'max_total_size', 'iou_threshold',
+    #       'score_threshold'
+    #   ]
+    # }
+
+
+if __name__ == "__main__":
+  test.main()
```

Comment:
Need to change the BUILD file for this, right?

Reply:
Done